### PR TITLE
usage example of i18n/smartling integration

### PR DESCRIPTION
**Pull request by bhainesva**
Process for managing translations:
* Project settings are configured in smartling.config.js. We should start generating new userID/userSecrets per client so that if we have to transition the repo to a partner we can decommission the key without causing problems. 
* Locales are also specified in `smartling.config.js` it's important that they're exported because they're also used in the configuration of some other tooling.
* To scan the repo for strings to translate, run `npx i18next`
* To push files to smartling to translate run `node i18n/index.js push`
* To pull translations from smartling run `node i18n/index.js pull`

We can add convenience scripts to `package.json` so that they're easier to run, but that's what's happening under the hood.

### CODE REVIEW

File: src/layouts/entity.tsx [hunk 1]
```diff
@@ -56,8 +59,15 @@ const EntityLayout = ({ data }: EntityLayoutProps) => {
...
   const showReviews = reviews?.title && reviews?.reviews;
 
+  const { t } = useTranslation();
+  const count = Math.ceil(Math.random() * 2);
+
   return (
     <>
+      <Trans>Welcome to My House</Trans>
+      <h1>{t("Welcome to React")}</h1>
+      <Trans count={count}>You have {{ count }} messages.</Trans> (Refresh to
```
Comment (JRonkin) [inline src/layouts/entity.tsx:69]: Are these two use cases supported?
1. Using JSX in replacements: https://github.com/yextconsulting/pages-starter-react-consulting/compare/jronkin/i18n..jronkin/translations#diff-3d90ccc767139e6a4893cc2a8db9cce23e38307d12bd763c661d880c92651cf8R24
2. Using replacements in places where we can't use JSX, i.e. the `t("Text {{ count }}")` form -- would that just have to be `t("Text {{ count }}").replaceAll("{{ count }}", count)`?

Comment (bhainesva) [reply src/layouts/entity.tsx:84]: Added examples for each of these use cases.

File: locales/en/translation.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "Welcome to My House": "Welcome to My House",
+  "Translated content with a custom component: <2>translated link text</2>": "Translated content with a custom component: <2>translated link text</2>",
+  "Welcome to React": "",
+  "{{author}} is the author of {{book}}": "",
+  "You have {{count}} messages.": "You have {{count}} messages.",
+  "You have {{count}} messages._plural": "You have {{count}} messages."
```
Comment (JRonkin) [inline locales/en/translation.json:7]: How does plural work with this library? In this example, they're both "messages". Should the first be "message"?

File: src/layouts/entity.tsx [hunk 3]
```diff
@@ -56,8 +60,29 @@ const EntityLayout = ({ data }: EntityLayoutProps) => {
...
   const showReviews = reviews?.title && reviews?.reviews;
 
+  const { t } = useTranslation();
+  const count = Math.ceil(Math.random() * 2);
+
   return (
     <>
+      <Trans>Welcome to My House</Trans>
+      <div>
+        <Trans shouldUnescape={true}>
```
Comment (JRonkin) [inline src/layouts/entity.tsx:70]: Do we need to add shouldUnescape every time we have JSX in the string?

Comment (bhainesva) [reply src/layouts/entity.tsx]: No the prop isn't necessary, removed it.

File: locales/en/translation.json (hunk 2 above)
Comment (bhainesva) [reply locales/en/translation.json]: Yeah. The tool that generates the json files uses the value of the <Trans> component as the default translation. I forgot to do it, but you would update the 'translation' here to handle it appropriately. Updated in new commit.

Comment (JRonkin) [reply locales/en/translation.json]: @bhainesva Got it. How do we tell it that it should use the singular version when `count == 1` and the plural version in all other cases? Especially if we have more than one variable in the string, like `{{ count }} stores near {{ query }}`

Comment (bhainesva) [reply locales/en/translation.json]: @JRonkin `count` is a special variable specifically for use with plurals, and has to be passed as a prop to the `Trans` component. For other interpolations you don't need a prop and can use objects as described here: https://react.i18next.com/latest/trans-component#interpolation.

**Review by JRonkin — COMMENTED** [review]
I think this covers all use cases! Thanks Ben!

**Comment by bhainesva** [comment]
Closing since this isn't intended to be merged, just exists as an example.

**Pull request closed without merging**